pkg/ccl/backupccl/restore_job.go

```diff
@@ -335,6 +396,9 @@ func restore(
 					if mu.highWaterMark >= 0 {
 						d.Restore.HighWater = importSpans[mu.highWaterMark].Span.Key
 					}
+					if execCtx.ExecCfg().Settings.Version.IsActive(restoreCtx, clusterversion.V23_1Start) {
+						d.Restore.Frontier = mu.spanFrontierSlice
```

Comment:
see comment below about a different way of flushing progress, and the comment above about cluster version :D